packages/flutter/lib/src/material/i18n/README.md

```diff
@@ -0,0 +1,105 @@
+
+There is one language-specifc .arb file for all supported locales. If an additional file with a regional suffix is present, the regional localizations are merged with the language-specific ones.
+
+The JSON table's keys, called resource IDs, are valid Dart variable names. They correspond to methods from the `MaterialLocalizations` class. For example:
+
+```dart
+Widget build(BuildContext context) {
+  return new FlatButton(
+    child: new Text(
+      MaterialLocalizations.of(context).cancelButtonLabel,
+    ),
+  );
+}
+```
+
+This widget build method creates a button whose label is the local translation of "CANCEL" which is defined for the `cancelButtonLabel` resource ID.
+
+Each of the language-specific .arb files contains an entry for `cancelButtonLabel`. They're all represented by the Dart `Map` in the generated `localizations.dart` file. The Map is used by the MaterialLocalizations class.
+
+
+### material_en.arb Defines all of the resource IDs
+
+All of the `.arb` files whose names do not include a regional suffix contain translations for the same set of resource IDs as `material_en.arb`.
+
+For each resource ID defined for English in material_en.arb, there is an additional resource with an '@' prefix. These '@' resources are not used by the material library at run time, they just exist to inform translators about how the value will be used.
+
+```dart
+"cancelButtonLabel": "CANCEL",
+"@cancelButtonLabel": {
+  "description": "The label for cancel buttons and menu items.",
+  "type": "text"
+},
+```
+
+
+### Values with Parameters, Plurals
+
+A few of material translations contain `$variable` tokens. The material library replaces these tokens with values at run-time. For example:
+
+```dart
+"aboutListTileTitle": "About $applicationName",
+```
+
+The value for this resource ID is retrieved with a parameterized method instead of a simple getter:
+
+```dart
+MaterialLocalizations.of(context).aboutListTileTitle(yourAppTitle)
+```
+
+Plurals are handled similarly, with a lookup method that includes a quantity parameter. For example `selectedRowCountTitle` returns a string like "1 item selected" or "no items selected".
+
+```dart
+MaterialLocalizations.of(context).selectedRowCountTitle(yourRowCount)
+```
+
+Plural translations can be provided for several quantities: 0, 1, 2, "few", "many", "other". The variations are identified by a resource ID suffix which must be one of "Zero", "One", "Two", "Few", "Many", "Other". The "Other" variation is used when none of the other quantities apply. All plural resources must include a resource with the "Other" suffix. For example the English translations ('material_en.arb') for `selectedRowCountTitle` are:
+
+```dart
+"selectedRowCountTitleZero": "No items selected",
+"selectedRowCountTitleOne": "1 item selected",
+"selectedRowCountTitleOther": "$selectedRowCount items selected",
+```
```

Comment:
this contradicts the text earlier that says that english contains all the strings